parser-typechecker/src/Unison/Builtin/Decls.hs

```diff
@@ -115,6 +129,68 @@ builtinDataDecls = rs1 ++ rs
                      (var "a" `arr` Type.app' (var "Optional") (var "a"))
       )
     ]
+  eith = DataDeclaration
+    Structural
+    ()
+    [v "a", v "b"]
+    [ ( ()
+      , v "Either.Left"
+      , Type.foralls () [v "a", v "b"]
+          (var "a" `arr` Type.apps' (var "Either") [var "a", var "b"])
+      )
+    , ( ()
+      , v "Either.Right"
+      , Type.foralls () [v "a", v "b"]
+          (var "b" `arr` Type.apps' (var "Either") [var "a", var "b"])
+      )
+    ]
+  fmode = DataDeclaration
+    Structural
+    ()
+    []
+    [ ((), v "FileMode.Read", var "FileMode")
+    , ((), v "FileMode.Write", var "FileMode")
+    , ((), v "FileMode.Append", var "FileMode")
+    , ((), v "FileMode.ReadWrite", var "FileMode")
+    ]
+  bmode = DataDeclaration
+    Structural
+    ()
+    []
+    [ ((), v "BufferMode.NoBuffering", var "BufferMode")
+    , ((), v "BufferMode.LineBuffering", var "BufferMode")
+    , ((), v "BufferMode.BlockBuffering", var "BufferMode")
+    , ((), v "BufferMode.SizedBlockBuffering"
+      , Type.nat () `arr` var "BufferMode")
+    ]
+  ioerr = DataDeclaration
+    Structural
+    ()
+    []
+    [ ((), v "AlreadyExists", var "IOError")
+    , ((), v "NoSuchThing", var "IOError")
+    , ((), v "ResourceBusy", var "IOError")
+    , ((), v "ResourceExhausted", var "IOError")
+    , ((), v "EOF", var "IOError")
+    , ((), v "IllegalOperation", var "IOError")
+    , ((), v "PermissionDenied", var "IOError")
+    , ((), v "UserError", var "IOError")
+    ]
+  seqview = DataDeclaration
```

Comment:
Can you explain how this is used? 

I think it's equivalent to `SeqView a = Empty | SeqView a (SeqView a)` which is just a cons list.

I could see `SeqView a seq = Empty | SeqView a seq` which you might use for uncons-ing or unsnoc-ing.

Reply:
Okay, I made this basically `type SeqView a b = Empty | Elem a b`. This fixes something else that was bugging me too, because it gets used for both left and right viewing, so either side may be the sequence depending on which you do.

Reply:
Ah, nice! It's always bugged me that there's a separate `ViewL` and `ViewR` type in Haskell for like Data.Sequence.